core/src/layers/complete.rs

```
@@ -549,16 +563,24 @@ where
             Error::new(ErrorKind::Unexpected, "writer has been closed or aborted")
         })?;
 
-        w.write(bs)
+        let len = bs.len();
+        w.write(bs)?;
+        self.size += len as u64;
+
+        Ok(())
     }
 
-    fn close(&mut self) -> Result<()> {
+    fn close(&mut self) -> Result<Metadata> {
         let w = self.inner.as_mut().ok_or_else(|| {
             Error::new(ErrorKind::Unexpected, "writer has been closed or aborted")
         })?;
 
-        w.close()?;
+        let mut ret = w.close()?;
+        if ret.content_length() == 0 {
```

We can insert a check here, if the response returns a size but it's differ with what we write, we should raise an error.

---

It only works when it isn't  an `append` operation. Do we still need this check ?

---

The point here is to act as a safeguard against poor implementations. Although we do our best to implement services and review them, there is still a possibility that some services may be implemented incorrectly. This check helps detect errors early, allowing us to catch them during the CI process.